lib/lcdbwf/tests/testthat/test-dds.R

```diff
@@ -1,17 +1,119 @@
-test_that("stripping gene versions", {
+#library(DESeq2)
+#library(testthat)
+#devtools::load_all('../../../../lib/lcdbwf')
+config <- lcdbwf:::load_config('../../../../workflows/rnaseq/downstream/config.yaml')
+source('test-functions.R')
+
+# --------- Test strip_dotted_version_from_dds() ---------- #
+test_that("strip_dotted_version_from_dds works", {
   dds <- DESeq2::makeExampleDESeqDataSet()
   # genes are labeled like "gene1", "gene2", etc; change them to "gene1.1",
   # gene2.2", and so on
-
   rownames(dds) <- paste(rownames(dds), seq(1000), sep='.')
-
   expect_error(lcdbwf:::strip_dotted_version_from_dds(dds), "Gene names don't appear to be Ensembl")
-
   forced <- lcdbwf:::strip_dotted_version_from_dds(dds, force=TRUE)
   expect_equal(rownames(forced)[1], "gene1")
-
   rownames(dds) <- paste0("ENS", rownames(dds), '.', seq(1000))
   fixed <- lcdbwf:::strip_dotted_version_from_dds(dds)
   expect_equal(rownames(fixed)[1], "ENSgene1")
+}) # test_that
+# --------------------------------------------------------- #
+
+# --------------------- Test make_dds() ------------------- #
+test_that("make_dds handles minimum required design data of sampletale,
+           and design formula, using example featurecounts table", {
+  design_data <- make_design_data()
+  dds <- make_dds(design_data,
+                  config=config,
+                  featureCounts='featurecounts.txt',
+                  parallel=config$parallel$parallel)
+  # Check that the dds object is a DESeqDataSet (not NULL)
+  expect_true(inherits(dds, "DESeqDataSet"))
+  # Check that the WaldStatistic has been generated as Wald is the default test that
+  # is run with minimal input to make_dds()
+  expect_true(any(grepl('Wald',  colnames(mcols(dds)))))
+  expect_false(any(grepl('LRT',  colnames(mcols(dds)))))
+}) # test_that
+
+test_that("make_dds handles design data where Wald test is specified explicitly", {
+  design_data <- make_design_data()
+  design_data$test <- 'Wald'
+  dds <- make_dds(design_data,
+                  config=config,
+                  featureCounts='featurecounts.txt',
+                  parallel=config$parallel$parallel)
+  # Check that the dds object is a DESeqDataSet (not NULL)
+  expect_true(inherits(dds, "DESeqDataSet"))
+  # Check that the WaldStatistic has been generated when Wald is passed to make_dds()
+  expect_true(any(grepl('Wald',  colnames(mcols(dds)))))
+  expect_false(any(grepl('LRT',  colnames(mcols(dds)))))
+}) # test_that
+
+test_that("make_dds handles required design data for LRT", {
+  design_data <- make_design_data()
+  design_data$test <- 'LRT'
+  design_data$reduced_design <- ~1
+  dds <- make_dds(design_data,
+                  config=config,
+                  featureCounts='featurecounts.txt',
+                  parallel=config$parallel$parallel)
+  # Check that the dds object is a DESeqDataSet (not NULL)
+  expect_true(inherits(dds, "DESeqDataSet"))
+  # Check that the WaldStatistic has been generated
```

Comment:
```suggestion
  # Check that the WaldStatistic has not been generated
```

Come to think of it, the `expect_true` and `expect_false` are clear enough that you prob don't need comments for them at all.